### PR TITLE
File optimization

### PR DESCRIPTION
**Pull request by haiqi96**
# References
N/A

# Description
1.Removed two strings member variable from file object. The getter now directly calls boost api instead of keeping the temporary strings in the class.
2. Turned static allocated page allocator object into dynamically allocated ones. This allows the clp to free up more memory when marking a file as closed.

# Validation performed
Ran some local testing and verified that max RSS memory usage went down.

### CODE REVIEW

**Comment by kirkrodrigues** [comment]
Can you fill out the PR template? It serves as good documentation for anyone reviewing the PR later.

File: components/core/src/streaming_archive/writer/File.hpp [hunk 1]
```diff
@@ -58,7 +58,11 @@ namespace streaming_archive { namespace writer {
                 m_segmentation_state(SegmentationState_NotInSegment),
                 m_is_metadata_clean(false),
                 m_is_written_out(false),
-                m_is_open(false)
+                m_is_open(false),
+                m_timestamps(nullptr),
+                m_logtypes(nullptr),
+                m_variables(nullptr)
+
```
Comment (kirkrodrigues) [inline components/core/src/streaming_archive/writer/File.hpp:65]: Unnecessary newline

File: components/core/src/streaming_archive/writer/File.hpp [hunk 2]
```diff
@@ -58,7 +58,11 @@ namespace streaming_archive { namespace writer {
                 m_segmentation_state(SegmentationState_NotInSegment),
                 m_is_metadata_clean(false),
                 m_is_written_out(false),
-                m_is_open(false)
+                m_is_open(false),
+                m_timestamps(nullptr),
```
Comment (kirkrodrigues) [inline components/core/src/streaming_archive/writer/File.hpp:62]: Why the explicit initialization to null?

File: components/core/src/streaming_archive/writer/File.hpp [hunk 3]
```diff
@@ -147,9 +151,9 @@ namespace streaming_archive { namespace writer {
          */
         const std::string& get_orig_path () const { return m_orig_log_path; }
         const boost::uuids::uuid& get_orig_file_id () const { return m_orig_file_id; }
-        const std::string& get_orig_file_id_as_string () const { return m_orig_file_id_as_string; }
+        const std::string get_orig_file_id_as_string () const { return boost::uuids::to_string(m_orig_file_id); }
```
Comment (kirkrodrigues) [inline components/core/src/streaming_archive/writer/File.hpp:154]: For `get_orig_file_id_as_string` and `get_id_as_string`
* Can remove `const` from the return type
* If these methods exist, the member can be removed right?

File: components/core/src/GlobalMySQLMetadataDB.cpp [hunk 4]
```diff
@@ -148,10 +148,10 @@ void GlobalMySQLMetadataDB::update_metadata_for_files (const std::string& archiv
     }
     auto& statement_bindings = m_upsert_file_statement->get_statement_bindings();
     for (auto file : files) {
-        const auto& id_as_string = file->get_id_as_string();
+        const auto id_as_string = file->get_id_as_string();
```
Comment (kirkrodrigues) [inline components/core/src/GlobalMySQLMetadataDB.cpp:151]: In `MetadataDB::update_files` and `GlobalSQLiteMetadataDB::update_metadata_for_files`, I wrote calls like:

```
m_upsert_file_statement->bind_text(enum_to_underlying_type(FilesTableFieldIndexes::OrigFileId) + 1, file->get_orig_file_id_as_string(), false);
```

Passing `false` means that the method does not need to copy the string; instead, it expects it to live until the statement is run (I probably should've added a comment to make that clear). With the change in return type of `get_id_as_string` and `get_orig_file_id_as_string`, the lifetime of the string is no longer clear. So, just as you do here, I would save the return values in local variables in those two methods.

**Review by kirkrodrigues — CHANGES_REQUESTED** [review]

File: components/core/src/streaming_archive/writer/File.hpp (hunk 2 above)
Comment (haiqi96) [reply components/core/src/streaming_archive/writer/File.hpp:62]: I am not familiar with unique ptr. If I don't null initialize them, will they just be default initialized?

Comment (kirkrodrigues) [reply components/core/src/streaming_archive/writer/File.hpp:62]: Yep, they start out empty.

**Review by kirkrodrigues — APPROVED** [review]

**Pull request merged**